src/Microsoft.DotNet.Build.CloudTestTasks/SendJsonToDocumentDb.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    /// <summary>
+    /// Used to send JSON blobs to DocumentDb.
+    /// </summary>
+    public class SendJsonToDocumentDb : Task
+    {
+        /// <summary>
+        /// The account key used to connect to DocumentDB.
+        /// </summary>
+        [Required]
+        public string AccountKey { get; set; }
+
+        /// <summary>
+        /// The name of the document collection.
+        /// </summary>
+        [Required]
+        public string Collection { get; set; }
+
+        /// <summary>
+        /// The name of the document database.
+        /// </summary>
+        [Required]
+        public string Database { get; set; }
+
+        /// <summary>
+        /// The ID of the document to be created.
+        /// </summary>
+        [Required]
+        public string DocumentId { get; set; }
+
+        /// <summary>
+        /// The DocumentDB endpont URI.
+        /// </summary>
+        [Required]
+        public string EndpointUri { get; set; }
+
+        /// <summary>
+        /// The JSON file to be uploaded.
+        /// </summary>
+        [Required]
+        public string JsonFile { get; set; }
+
+        static SendJsonToDocumentDb()
+        {
+            AssemblyResolver.Enable();
+        }
+
+        public override bool Execute()
+        {
+            // load the JSON file
+            JObject json;
+
+            using (StreamReader streamReader = new StreamReader(JsonFile))
+            using (JsonTextReader jsonReader = new JsonTextReader(streamReader))
+            {
+                var jsonSerializer = new JsonSerializer();
+                json = (JObject)jsonSerializer.Deserialize(jsonReader);
+
+                // add the document ID to the JSON blob
+                json.Add("id", DocumentId);
+            }
+
+            using (var con = CreateDocumentDbConnection())
+            {
+                // for specified document ID check if there is a document for it, if there
+                // isn't then create a new one with the JSON blob that was loaded earlier.
+
+                var result = con.Client.CreateDocumentQuery(string.Format("dbs/{0}/colls/{1}", con.Database.Id, con.Collection.Id))
+                    .Where(doc => doc.Id == DocumentId)
+                    .AsEnumerable()
+                    .FirstOrDefault();
+
+                if (result == null)
+                {
+                    Log.LogMessage("Creating new document with ID '{0}'", DocumentId);
+                    using (var task = con.Client.CreateDocumentAsync(con.Collection.SelfLink, json))
```

Comment:
Since it implements IDisposable I figured it needed to be disposed.  I did a little more digging and found http://blogs.msdn.com/b/pfxteam/archive/2012/03/25/10287435.aspx, apparently it's not needed in most cases.